Content.Server/GameTicking/Rules/RevolutionaryRuleSystem.cs

```diff
@@ -0,0 +1,463 @@
+    [Dependency] private readonly PopupSystem _popup = default!;
+    [Dependency] private readonly RoundEndSystem _roundEndSystem = default!;
+    [Dependency] private readonly SharedAudioSystem _audioSystem = default!;
+    [Dependency] private readonly SharedChargesSystem _charges = default!;
+    [Dependency] private readonly SharedStunSystem _sharedStun = default!;
+    [Dependency] private readonly StationSpawningSystem _stationSpawningSystem = default!;
+    [Dependency] private readonly StationJobsSystem _stationJobs = default!;
+    [Dependency] private readonly StationSystem _stationSystem = default!;
+    [Dependency] private readonly StorageSystem _storageSystem = default!;
+
+    private TimeSpan _timerWait = TimeSpan.FromSeconds(20);
+
+    private TimeSpan _endRoundCheck = default!;
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<RoundStartAttemptEvent>(OnStartAttempt);
+        SubscribeLocalEvent<RulePlayerJobsAssignedEvent>(OnPlayerJobAssigned);
+        SubscribeLocalEvent<PlayerSpawnCompleteEvent>(OnPlayerSpawned);
+        SubscribeLocalEvent<RevolutionaryRuleComponent, MobStateChangedEvent>(OnMobStateChanged);
+        SubscribeLocalEvent<RoundEndTextAppendEvent>(OnRoundEndText);
+        SubscribeLocalEvent<HeadRevolutionaryComponent, AfterFlashedEvent>(OnPostFlash);
+    }
+
+    protected override void Started(EntityUid uid, RevolutionaryRuleComponent component, GameRuleComponent gameRule, GameRuleStartedEvent args)
+    {
+        base.Started(uid, component, gameRule, args);
+        _endRoundCheck = _timing.CurTime + _timerWait;
+    }
+
+    /// <summary>
+    /// Checks if the round should end and also checks who has a mindshield.
+    /// </summary>
+    protected override void ActiveTick(EntityUid uid, RevolutionaryRuleComponent component, GameRuleComponent gameRule, float frameTime)
+    {
+        base.ActiveTick(uid, component, gameRule, frameTime);
+        if (GameTicker.IsGameRuleAdded(uid, gameRule) && _endRoundCheck <= _timing.CurTime)
+        {
+            _endRoundCheck = _timing.CurTime + _timerWait;
+            CheckFinish();
+        }
+    }
+
+    private void OnRoundEndText(RoundEndTextAppendEvent ev)
+    {
+        foreach (var headrev in EntityQuery<RevolutionaryRuleComponent>())
+        {
+            if (headrev.HeadsDied && !headrev.RevsLost)
+            {
+                ev.AddLine(Loc.GetString("rev-won"));
+            }
+            if (!headrev.HeadsDied && headrev.RevsLost)
+            {
+                ev.AddLine(Loc.GetString("rev-lost"));
+            }
+            if (headrev.HeadsDied && headrev.RevsLost)
+            {
+                ev.AddLine(Loc.GetString("rev-stalemate"));
+            }
+            if (!headrev.HeadsDied && !headrev.RevsLost)
+            {
+                ev.AddLine(Loc.GetString("rev-reversestalemate"));
+            }
```

Comment:
Use a bitmask rather than checking 2 separate bools, also if these are exclusive use a switch or if-else statements.

Reply:
I had to look up bitmasks to see what it was and I'm gonna keep it real, I don't think I'm smart enough to understand how a bitmask works.

Reply:
it's just an enum of options.

Reply:
This shit not clicking either